packages/flutter/lib/src/widgets/widget_inspector.dart

```diff
@@ -3646,3 +3646,64 @@ class InspectorSerializationDelegate implements DiagnosticsSerializationDelegate
     );
   }
 }
+
+class _WidgetFactory {
+  const _WidgetFactory();
+}
+
+/// Annotation which marks a function as a widget factory for the purpose of
+/// widget creation tracking.
+///
+/// When widget creation tracking is enabled, the framework tracks the source
+/// code location of the constructor call for each widget instance. This
+/// information is used by the DevTools to provide an improved developer
+/// experience. For example, it allows the Flutter inspector to present the
+/// widget tree in a manner similar to how the UI was defined in your source
+/// code.
+///
+/// [Widget] constructors are automatically instrumented to track the source
+/// code location of constructor calls. However, there are cases where
+/// a function acts as a sort of a constructor for a widget and a call to such
+/// a function should be considered as the creation location for the returned
+/// widget instance.
+///
+/// Annotating a function with this annotation marks the function as a widget
+/// factory. The framework will then instrument that function in the same way
```

Comment:
Where does this instrumentation happen? Where is that implemented? As far as I can tell, this PR just defines an annotation that isn't doing anything?

Reply:
I mentioned it in the PR description, but maybe not clearly enough :)

The instrumentation is implemented in the CFE kernel transformer that also implements the instrumentation of all classes extending `Widget`. This is the commit that landed support for widget factories in the Dart SDK: https://github.com/dart-lang/sdk/commit/fe3f2188ffecbdbfb0b4b2ed587ddcb92ba57fcd.

Reply:
Thanks for the clarifications!